static/app/views/relocation/uploadBackup.tsx

```diff
@@ -0,0 +1,248 @@
+import {useContext, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+import {motion} from 'framer-motion';
+
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {Client} from 'sentry/api';
+import {Button} from 'sentry/components/button';
+import {RelocationOnboardingContext} from 'sentry/components/onboarding/relocationOnboardingContext';
+import Well from 'sentry/components/well';
+import {IconFile, IconUpload} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import {space} from 'sentry/styles/space';
+import testableTransition from 'sentry/utils/testableTransition';
+import useApi from 'sentry/utils/useApi';
+import StepHeading from 'sentry/views/relocation/components/stepHeading';
+
+import {StepProps} from './types';
+
+type UploadWellProps = {
+  centered: boolean;
+  draggedOver: boolean;
+  onDragEnter: Function;
+  onDragLeave: Function;
+  onDragOver: Function;
+  onDrop: Function;
+};
+
+export function UploadBackup(__props: StepProps) {
+  const api = useApi({
+    api: new Client({headers: {Accept: 'application/json; charset=utf-8'}}),
+  });
+  const [file, setFile] = useState<File>();
+  const [dragCounter, setDragCounter] = useState(0);
+  const inputFileRef = useRef<HTMLInputElement>(null);
+  const relocationOnboardingContext = useContext(RelocationOnboardingContext);
+  const user = ConfigStore.get('user');
+
+  const handleDragEnter = event => {
+    event.preventDefault();
+    setDragCounter(dragCounter + 1);
+  };
+
+  const handleDragLeave = () => {
+    setDragCounter(dragCounter - 1);
+  };
+
+  const handleDrop = event => {
+    event.preventDefault();
+    setDragCounter(0);
+
+    setFile(event.dataTransfer.files[0]);
+  };
+
+  const handleFileChange = event => {
+    const newFile = event.target.files?.[0];
+
+    // No file selected (e.g. user clicked "cancel")
+    if (!newFile) {
+      return;
+    }
+
+    setFile(newFile);
+  };
+
+  const onFileUploadLinkClick = () => {
+    inputFileRef.current && inputFileRef.current.click();
+  };
+
+  const handleStartRelocation = async () => {
+    const {orgSlugs, regionUrl} = relocationOnboardingContext.data;
+    if (!orgSlugs || !regionUrl || !file) {
+      addErrorMessage(
+        t(
+          'An error has occured while trying to start relocation job. Please contact support for further assistance.'
+        )
+      );
+      return;
+    }
+    const formData = new FormData();
+    formData.set('orgs', orgSlugs);
+    formData.set('file', file);
+    formData.set('owner', user.email);
+    try {
+      await api.requestPromise(`/relocations/`, {
+        method: 'POST',
+        host: regionUrl,
+        data: formData,
+      });
+
+      addSuccessMessage('Triggered Relocation Job');
+    } catch (error) {
+      addErrorMessage(error.responseJSON?.detail);
```

Comment:
nit: I don't think we should show them the details of the error message by default. Looking at the POST endpoint code in `.../api/endpoints/relocations/index.py`, we have 3 broad cases of errors:

- 409: This person has already started a relocation
- 429: We've reached the daily limit on relocations
- 401: Their session has expired.
- 400: All other sorts of errors.

I believe we should match on error code and show custom messages for each of the first 3, since those are ephemeral or fixable by the user. We should also add tests for each.

For the 400 error code, and any other error code besides, the screw up is almost certainly on our side: the owning account does not exist, relocation has been disabled as a feature, some other invariant between the way our front and back ends interact has been violated, etc. In this case, we should just throw up a generic "Something went wrong - please contact support" type of error message, since giving them the gory details of what is essentially programmer error is more confusing than anything, and we'll have enough info in the logs to debug it anyway.

Reply:
good point

Reply:
done